show/main.py

```diff
@@ -119,7 +121,20 @@ def connect_config_db():
     config_db.connect()
     return config_db
 
+def is_gearbox_configured():
+    """
+    Checks whether Gearbox is configured or not
+    """
+    app_db = SonicV2Connector(host='127.0.0.1')
+    app_db.connect(app_db.APPL_DB)
 
+    state_record = app_db.get(app_db.APPL_DB, GEARBOX_TABLE_STATE_RECORD, "success")
```

Comment:
I have double checked https://github.com/Azure/SONiC/blob/b16791f3ca16398153556f40b11831cfbb6aba79/doc/gearbox/gearbox_mgr_design.md and implementation of Gearsyncd. I found that "GearboxConfigDone" is being set in any cases, even when the system has no gearbox(in this case "success" record is always 1, see https://github.com/Azure/sonic-swss/blob/321291aa7aca89a3ccef2bed36aff074eac1054f/gearsyncd/gearsyncd.cpp#L91).
So, I think there is no necessity to make "app_db.exists()", because "GearboxConfigDone" record should be always present in the DB.
The only case, in which "GearboxConfigDone" record will be absent, is if there is something wrong with Gearsyncd, e.g. if some defect has made inside the daemon or any related part. In this case, in my opinion, it would be better to see the warnings to speedup detection of the defect.